src/sentry/api/endpoints/project_symbol_sources.py

```diff
@@ -0,0 +1,386 @@
+        min_length=36,
+        max_length=36,
+        required=False,
+        help_text="The [App Store Connect Issuer ID](https://developer.apple.com/documentation/appstoreserverapi/generating_tokens_for_api_requests). Required for AppStoreConnect sources, invalid for all others.",
+    )
+    appconnectKey = (
+        serializers.CharField(
+            min_length=2,
+            max_length=20,
+            required=False,
+            help_text='The [App Store Connect API Key](https://developer.apple.com/documentation/appstoreconnectapi/creating_api_keys_for_app_store_connect_api) ID. Note that the key must have the "Developer" role for Sentry to discover the app builds. Required for AppStoreConnect sources, invalid for all others.',
+        ),
+    )
+    appconnectPrivateKey = serializers.CharField(
+        required=False,
+        help_text="The [App Store Connect API Private Key](https://developer.apple.com/documentation/appstoreconnectapi/creating_api_keys_for_app_store_connect_api). Required for AppStoreConnect sources, invalid for all others.",
+    )
+    appName = (
+        serializers.CharField(
+            min_length=1,
+            max_length=512,
+            required=False,
+            help_text="The [App Store Connect App Name](https://developer.apple.com/help/app-store-connect/create-an-app-record/add-a-new-app). Required for AppStoreConnect sources, invalid for all others.",
+        ),
+    )
+    appId = serializers.CharField(
+        min_length=1,
+        required=False,
+        help_text="The App Store Connect App ID. Required for AppStoreConnect sources, invalid for all others.",
+    )
+    bundleId = (
+        serializers.CharField(
+            min_length=1,
+            required=False,
+            help_text="The [App Store Connect App Bundle](https://developer.apple.com/help/app-store-connect/create-an-app-record/create-and-submit-app-bundles) ID. Required for AppStoreConnect sources, invalid for all others.",
+        ),
+    )
+    url = serializers.CharField(
+        required=False,
+        help_text="The source's URL. Optional for HTTP sources, invalid for all others.",
+    )
+    username = serializers.CharField(
+        required=False,
+        help_text="The user name for accessing the source. Optional for HTTP sources, invalid for all others.",
+    )
+    password = serializers.CharField(
+        required=False,
+        help_text="The password for accessing the source. Optional for HTTP sources, invalid for all others.",
+    )
+    bucket = serializers.CharField(
+        required=False,
+        help_text="The GCS or S3 bucket where the source resides. Required for GCS and S3 sourcse, invalid for HTTP and AppStoreConnect sources.",
+    )
+    region = serializers.CharField(
+        required=False,
+        help_text="The source's [S3 region](https://docs.aws.amazon.com/general/latest/gr/s3.html). Required for S3 sources, invalid for all others.",
```

Comment:
Do we need to do any validation for `region`? I noticed we're not comparing it to a set list anywhere, and we only validate against the JSON which just checks that the input is a string.

Reply:
In the UI this is a dropdown, so I think it would make sense to restrict this choice here as well.